.github/workflows/ros2.yml

```diff
@@ -1,28 +1,37 @@
 name: ROS2 CI
-on:
+
+on: # this determines when this workflow is run
   push:
+    branches:
+      - ros2
   pull_request:
-  schedule:
-    - cron: "0 0 * * *"
+  workflow_dispatch: # allow manually starting this workflow
+
 jobs:
   industrial_ci:
-    strategy:
-      matrix:
-        env:
-          - ROS_DISTRO: foxy
-            ROS_REPO: testing
-            CMAKE_ARGS: '-DCMAKE_BUILD_TYPE=Debug'
-          - ROS_DISTRO: foxy
-            ROS_REPO: testing
-            CMAKE_ARGS: '-DCMAKE_BUILD_TYPE=Release'
-          - ROS_DISTRO: galactic
-            ROS_REPO: testing
-            CMAKE_ARGS: '-DCMAKE_BUILD_TYPE=Debug'
-          - ROS_DISTRO: galactic
-            ROS_REPO: testing
-            CMAKE_ARGS: '-DCMAKE_BUILD_TYPE=Release'
+    name: ROS ${{ matrix.ROS_DISTRO }} (${{ matrix.ROS_REPO }})
     runs-on: ubuntu-latest
+    strategy:
+      # fail-fast: false # uncomment if failing jobs should not cancel the others immediately
+      matrix: # matrix is the product of entries
+        ROS_DISTRO: [humble, iron, rolling]
+        ROS_REPO: [testing, main]
+    env:
+      CCACHE_DIR: "${{ github.workspace }}/.ccache" # directory for ccache (and how we enable ccache in industrial_ci)
     steps:
-      - uses: actions/checkout@v1
-      - uses: 'ros-industrial/industrial_ci@master'
-        env: ${{matrix.env}}
+      - uses: actions/checkout@v3 # clone target repository
+      - uses: actions/cache@v2 # fetch/store the directory used by ccache before/after the ci run
```

Comment:
For CI, I don want to use `cache`, can you remove this?
I want to use normal compiler to test the behavior